R/mz_summary.R

```diff
@@ -0,0 +1,185 @@
+      as.data.frame()
+    
+    sum_transform <- x_df %>% 
+      dplyr::group_by(combine_transformation) %>% 
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()  %>% 
+      tibble::add_column(sum_mass$combine_mass_difference)
+  }
+  
+  # if AdjacencyMatrix from `structural` is used 
+  else {
+    x_df <- x_df[x_df$binary == 1, ]
+    
+    sum_mass <- x_df %>% 
+      dplyr::group_by(mass_difference) %>%  
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()
+    
+    sum_transform <- x_df %>% 
+      dplyr::group_by(transformation) %>% 
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()  %>% 
+      tibble::add_column(sum_mass$mass_difference)
+  }
+  
+  colnames(sum_transform) <- c("transformation", "counts", "mass_difference")
+  sum_transform <- sum_transform %>% 
+    dplyr::select(transformation, mass_difference, counts)
+  
+  if (filter == FALSE | filter <= 1) {
+    return(sum_transform)
+  }
+  
+  else {
+    sum_f <- filter(sum_transform,sum_transform$counts >= filter)
+    return(sum_f)
+    
+  }
+  
+  
+}
+
+
+
+
+#' @name mz_vis
+#'
+#' @aliases mz_vis
+#'
+#' @title Visualize mass-difference distribution 
+#'
+#' @description
+#' The function `mz_vis` visualizes the mass-difference distribution,
+#' which has been summarized by `mz_summary`. 
+#' 
+#' @param x
+#' `list`, previously generated by `mz_summary`. Needs to contain
+#' the columns "transformation", "mass_difference" and "counts".
+#' 
+#' 
+#' @details
+#' Plots the mass-difference distribution, summarized
+#' by `mz_summary`. 
+#' Visualization is performed using ggplot2
+#' 
+#' @return 
+#' `ggplot` object and corresponding barplot for visualizations
+#'
+#' @author Liesa Salzer, \email{liesa.salzer@@helmholtz-muenchen.de}
+#'
+#' @examples
+#' data("x_test", package = "MetNet")
+#' transformation <- rbind(
+#'     c("Monosaccharide (-H2O)", "C6H10O5", "162.0528234315"),
+#'     c("Disaccharide (-H2O)", "C12H20O11", "340.1005614851"),
+#'     c("Trisaccharide (-H2O)", "C18H30O15", "486.1584702945"))
+#' transformation <- data.frame(group = transformation[, 1],
+#'                                 formula = transformation[, 2],
+#'                                 mass = as.numeric(transformation[, 3]))
+#' am_struct <- structural(x_test, transformation, ppm = 5, directed = TRUE)
+#' mz_sum <- mz_summary(am_struct)
+#' mz_vis(mz_sum)
+#'
+#'@export
+mz_vis <- function(x, ...){
+   
+  if (!all(c("transformation", "mass_difference", "counts") %in% colnames(x))) 
+    stop("'x' has not the right columns 'transformation', 'mass_difference', 'counts'")  
+  
+    plot_list <- ggplot2::ggplot(x, 
+                                 ggplot2::aes_string(x="transformation", y="counts")) + 
+      ggplot2::geom_bar(stat = "identity") + 
+      ggplot2::theme_minimal() + 
+      ggplot2::coord_flip() + 
+      ggplot2::labs(title = "Numbers of determined mass differences")
+    
+    plot(plot_list)
+  
```

Comment:
is there any specific reason why to assign the `ggplot2::...` expression to `plot_list` and use then `plot(plot_list)`? Why not having only the `ggplot2::...` expression here?